### ISSUE TITLE
Improvals to function calls

### ISSUE THREAD

**Issue by Try** [GothicKit/ZenKit#81]
Some issues with functions
1. Scripts compiled by vanilla compiler may have inconsistent sequence of `push`/`pop`, leaving more  data in stack, that expected by caller
Proposal: improve guard-related code in `opcode::bl`

2. Need to have traps for call-exit. Two use-cases here:
a) exit from loop-body:
```
BUCK = _^(BUCKET);
REPEAT(I, (BUCK.NUMINARRAY) / (2)); // opengothic tracks repeat/while loops and memorizes pointer to I, num iterations
if ((MEM_ARRAYREAD(BUCKET, (I) * (2))) == (KEY)) {
    return MEM_ARRAYREAD(BUCKET, ((I) * (2)) + (1)); // ...and tracking fails here
};
END;
```
b) LeGo locals package: locals allow to annotate some functions to have proper local-variables. On side of opengothic I can trap annotation just fine and stash 'local' variables, but need to have a way to restore previous values on function exit

PS:
can do it after finishing with memory-traps PR :)

**Comment by lmichaelis** [comment]
> Tested vanilla G2-notr, via:

Yes I know these cases exist, but where does extra data on the stack cause actual problems?

> Are you sure? Hard to imagine case, when corrupted stack is desired way of running the script.

It's more of a thing about accurately running the code. For example, it is conceivable that someone might implement a compiler with support for multiple return types in which case this change would break their implementation though they did generate valid Daedalus bytecode. There are probably other cases in which things might break too.

**Comment by Try** [comment]
> Yes I know these cases exist, but where does extra data on the stack cause actual problems?

Lack of data - problem in CoM. Getting `"popping instance from empty stack"` message.
Extra data, in theory - yes, if something like `foo(boo(), ext_data())` - in such case `boo` output would be overridden.

 > It's more of a thing about accurately running the code.

OK, will do with flag. But dough that compiler like that will come any time soon ;)
